### PR TITLE
Reduce timeouts for non-suspend ConfigServer REST API calls from node admin

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/ConfigServerApiImpl.java [hunk 1]
```diff
@@ -202,24 +209,34 @@ private static CloseableHttpClient createClient(SSLConnectionSocketFactory socke
...
                 .build();
     }
 
+    private static void setRequestConfigOverride(Params paramsOrNull, HttpRequestBase request) {
+        if (paramsOrNull == null) return;
+
+        RequestConfig.Builder builder = RequestConfig.copy(request.getConfig());
+
+        paramsOrNull.getConnectionTimeout().ifPresent(connectionTimeout -> {
+            builder.setConnectTimeout((int) connectionTimeout.getSeconds());
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/ConfigServerApiImpl.java]: Shouldn't this be millis?

**Comment by hakonhall** [comment]
Decided against increasing the timeout for suspend, which might hide the problem observed on controller-cd-2 on November 12. Instead, this PR allows tuning the timeouts of suspend independent of timeout used for other ConfigServer REST calls. And reduces current non-suspend from 15s to 10s (and getting connection from the 200-connection pool to 1s).

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/ConfigServerApi.java [hunk 2]
```diff
@@ -9,13 +10,26 @@
...
+        public Optional<Duration> getConnectionTimeout() { return connectionTimeout; }
+    }
 
     <T> T get(String path, Class<T> wantedReturnType);
 
     <T> T post(String path, Object bodyJsonPojo, Class<T> wantedReturnType);
 
     <T> T put(String path, Optional<Object> bodyJsonPojo, Class<T> wantedReturnType);
 
+    <T> T put(Params params, String path, Optional<Object> bodyJsonPojo, Class<T> wantedReturnType);
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/ConfigServerApi.java]: Consider putting the optional argument at the end

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/ConfigServerApiImpl.java (hunk 1 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/ConfigServerApiImpl.java]: Great catch - I just verified this but forgot to update the code...

**Review by freva — APPROVED** [review]
:100:

**Pull request merged**